platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -323,14 +352,76 @@ - (MGLImage *)blurredAttributionBackground:(CIImage *)backgroundImage
     CGImageRef cgimg = [ctx createCGImage:blurredImage fromRect:[backgroundImage extent]];
     
 #if TARGET_OS_IPHONE
-    
     return [UIImage imageWithCGImage:cgimg];
 #else
-    
     return [[NSImage alloc] initWithCGImage:cgimg size:[backgroundImage extent].size];
 #endif
 }
 
+- (MGLSnapshotAttributionOptions *)attributionOptionsForSize:(CGSize)snapshotSize attributionInfo:(NSArray *)attributionInfo
+{
+    NSMutableArray *options = [NSMutableArray array];
+    MGLSnapshotAttributionOptions *largeLogoAttribution = [self attributionOptionsForAttributionInfo:attributionInfo abbreviated:NO];
+#if TARGET_OS_IPHONE
+    largeLogoAttribution.logoImage = [UIImage imageNamed:@"mapbox" inBundle:[NSBundle mgl_frameworkBundle] compatibleWithTraitCollection:nil];
+#else
+    largeLogoAttribution.logoImage = [[NSImage alloc] initWithContentsOfFile:[[NSBundle mgl_frameworkBundle] pathForResource:@"mapbox" ofType:@"pdf"]];
+#endif
+    _defaultLogoHeight = largeLogoAttribution.logoImage.size.height;
+    
+    MGLSnapshotAttributionOptions *smallLogoAttribution = [self attributionOptionsForAttributionInfo:attributionInfo abbreviated:NO];
+#if TARGET_OS_IPHONE
+    smallLogoAttribution.logoImage = [UIImage imageNamed:@"mapbox_helmet" inBundle:[NSBundle mgl_frameworkBundle] compatibleWithTraitCollection:nil];
+#else
+    smallLogoAttribution.logoImage = [[NSImage alloc] initWithContentsOfFile:[[NSBundle mgl_frameworkBundle] pathForResource:@"mapbox_helmet" ofType:@"pdf"]];
+#endif
+    
+    MGLSnapshotAttributionOptions *noLogoAttribution = [self attributionOptionsForAttributionInfo:attributionInfo abbreviated:YES];
+    
+    [options addObject:largeLogoAttribution];
+    [options addObject:smallLogoAttribution];
+    [options addObject:noLogoAttribution];
+    
+    for (MGLSnapshotAttributionOptions *attributionOptions in options) {
+        // -[Mapbox Logo]-[Attribution Background]-
+        CGFloat origin = attributionOptions.logoImage ? MGLLogoImagePosition.x : 0;
+        CGFloat width = origin + attributionOptions.logoImage.size.width + 10 + attributionOptions.attributionBackgroundSize.width + 10;
+        if (width <= snapshotSize.width) {
+            return attributionOptions;
+        }
+    }
+    
+    return noLogoAttribution;
+}
+
+- (MGLSnapshotAttributionOptions *)attributionOptionsForAttributionInfo:(NSArray *)attributionInfo abbreviated:(BOOL)abbreviated
+{
+    NSString *openStreetMap = NSLocalizedStringWithDefaultValue(@"OPEN_STREET_MAP_ATTRIBUTION", nil, nil, @"OpenStreetMap", @"Open Street Map full name attribution");
+    NSString *OSM = NSLocalizedStringWithDefaultValue(@"OSM_ATTRIBUTION", nil, nil, @"OSM", @"Open Street Map short name attribution");
```

Comment:
“OpenStreetMap” has no spaces. The keys should be `OSM_LONG_NAME` and `OSM_SHORT_NAME`, respectively, and the comments should be “Full/abbreviated name of OpenStreetMap in attribution text.”